### ISSUE TITLE
[$50] When trying to connect to zoom, an incorrect interview time is shown 

### ISSUE THREAD

**Issue by sdgun** [topcoder-platform/taas-app#593]
Steps
Schedule an interview
Go to JC's email
Click the zoom link in the Candidate confirmed meeting on... email 
Check the time

 incorrect interview time is shown 

All Interviews here: https://platform.topcoder-dev.com/taas/myteams/19320/positions/c8511efd-af2d-46ee-a86d-59d291bee61e/candidates/interviews

https://user-images.githubusercontent.com/4592712/145806325-192532ee-41fb-416b-937d-7f5acbe58c90.mp4

**Comment by maxceem** [comment]
@sdgun just to confirm. `Asia/Colombo` is your current timezone when testing, right?
So when it says interview is scheduled for 2pm Asia/Colombo you are expecting that Zoom would also show the same `2pm` because you are now in `Asia/Colombo` timezone, right?

**Comment by sdgun** [comment]
> @sdgun just to confirm. `Asia/Colombo` is your current timezone when testing, right? So when it says interview is scheduled for 2pm Asia/Colombo you are expecting that Zoom would also show the same `2pm` because you are now in `Asia/Colombo` timezone, right?

@maxceem yeah, I was expecting to see 2pm.

**Comment by xxcxy** [comment]
Created a MR https://github.com/topcoder-platform/taas-apis/pull/644 to fix this.

**Comment by maxceem** [comment]
@sdgun this issue has been fixed and is ready for testing on DEV.

Here is a comment regarding the fix:
> A zoom meeting can only set one timezone when creating, and the only join url will show the waiting page, so I set the meeting timezone to the guest timezone.

**Comment by maxceem** [comment]
@xxcxy it looks like this issue is not yet fixed, could you please have a look https://github.com/topcoder-platform/taas-app/issues/593#issuecomment-995625140

**Comment by xxcxy** [comment]
Sure, I will check it. I want to confirm is the new meeting is fine? looks the video is a rescheduled meeting.

**Comment by sdgun** [comment]
> Sure, I will check it. I want to confirm is the new meeting is fine? looks the video is a rescheduled meeting.

The new meetings has the same issue @xxcxy

**Comment by xxcxy** [comment]
Sorry, I don't understand. I check it and looks it show correct. https://monosnap.com/file/ROrXOUIFzX3gAIwxJHd56mDIzksd3O

**Comment by xxcxy** [comment]
In the second video, you computer timezone is different from email timezone. The zoom client will show the time from your computer timezone.

**Comment by sdgun** [comment]
> In the second video, you computer timezone is different from email timezone. The zoom client will show the time from your computer timezone.

@xxcxy 
There is no TZ named 'Asia/colombo time zone' in windows, what I use is 'Sri Jayawardenapura' which is also 
 GMT+5:30(same as IST)

**Comment by xxcxy** [comment]
I'm not sure whether  it's the zoom client issue, could you try `Asia/Kolkata` timezone, if windows has timezone like this.

**Comment by sdgun** [comment]
@xxcxy I checked by changing the TZ to Asia/Kolkata, still see a different time


https://user-images.githubusercontent.com/4592712/146625943-24b55331-fea1-41a7-a004-df8fba28fb77.mp4

**Comment by xxcxy** [comment]
My last MR, I forgot to update the reschedule code, so the reschedule process may have timezone problem. I fix it in this MR https://github.com/topcoder-platform/taas-apis/pull/646 .
But this just determine the timezone of start time we send to zoom. How to show this time depends on zoom client. I tried `Asia/Shanghai` and `Europe/London` in my computer and it shows correctly.

**Comment by maxceem** [comment]
> But this just determine the timezone of start time we send to zoom. How to show this time depends on zoom client. I tried Asia/Shanghai and Europe/London in my computer and it shows correctly.

hmm, @xxcxy, so what you are saying is, that we actually don't have to use any specific timezone when creating a zoom meeting? We just can avoid setting timezone, pass time in UTC+0, and it should adjust to the client timezone of the one who opens the link?

**Comment by maxceem** [comment]
As from Zoom API docs:

![image](https://user-images.githubusercontent.com/146016/146670846-6654dd63-0321-48c8-9aa4-a845c86c02cf.png)

So it looks like we just have to pass the time in the correct format `yyyy-MM-ddTHH:mm:ssZ` which means GMT 0, and Zoom client would show the correct date depend on the computer timezone.

Is that right?

**Comment by xxcxy** [comment]
Yes, I tried if I set different timezone in my computer and restart the zoomMeeting client, it shows different time like the doc said. But if i don't set timezone when sending api, the client shows the time different from the email even they have same timezone. So i think we should set the timezone.

**Comment by maxceem** [comment]
Ok. So I guess **sdgun** still can reproduce the issue because in this:

![image](https://user-images.githubusercontent.com/146016/146672478-ac67d174-41ae-43b4-93be-61acfe62c0b2.png)

But it seems that we there could be easier solution then using candidate timezone:
- we can set time in UTC using specific time format `yyyy-MM-ddTHH:mm:ss`
- and set timezone as `UTC`

And then the client would render the time we set in UTC using computer timezone. Would this work?

**Comment by xxcxy** [comment]
No, I tried "UTC", at least it can't work in my computer. The time showed different from email.

**Comment by maxceem** [comment]
Hmm, could it be that `toISOString` returns date not in the same format that is expected by Zoom?
- `2011-10-05T14:48:00.000Z`
- if we set timezone, then there should not be `.000Z`

**Comment by xxcxy** [comment]
I'm not sure. Since I can't reproduce it in my computer and my zoomAccount.
I need confirm something to infer what's the problem.
1. if sdgun change the computer timezone, the zoom client shows different time?
2. what the actual time sdgun set and whether the email show time is right?

**Comment by maxceem** [comment]
@sdgun there are some details which might help to find the root cause of the issue https://github.com/topcoder-platform/taas-app/issues/593#issuecomment-997373702.

I guess a video showing the whole process including scheduling, selecting time and opening the final zoom link might help.

**Comment by sdgun** [comment]
@maxceem @xxcxy Here is the video 

Uploading Interview zoom time.mp4…

**Comment by sdgun** [comment]
@xxcxy Looks like the time is shown in GMT in Nylas and zoom, for example in the timeslots payload, the starttime and endtime stamps are sent when the user schedules the interview, for example in the below interview, JC scheduled the interview to 1pm-1.30pm on 25th Dec in IST, which is 7.30am-8am in GMT and that is what the Nylas/zoom shows to the JC while trying to connect the meeting.


https://user-images.githubusercontent.com/4592712/146877252-a9218514-331a-4dc8-a23c-f607b613fff1.mp4

**Comment by xxcxy** [comment]
Yes. I think the request parameters is very helpful, I will try to reproduce this issues in my local, If i can reproduce it, i think i can resolve it.

**Comment by xxcxy** [comment]
> Hmm, could it be that `toISOString` returns date not in the same format that is expected by Zoom?
> 
> * `2011-10-05T14:48:00.000Z`
> * if we set timezone, then there should not be `.000Z`

Yes, You are right.
Fixed in this MR https://github.com/topcoder-platform/taas-apis/pull/649

**Comment by maxceem** [comment]
@sdgun a new fix has been deployed to DEV and is ready for testing.

**Comment by sdgun** [comment]
Unable to verify due to https://github.com/topcoder-platform/taas-app/issues/596

**Comment by sdgun** [comment]
Verified in dev env, the time is displayed correctly now.


https://user-images.githubusercontent.com/4592712/147059823-6ed47db8-4dfa-48ad-b4af-b7d5dab051c2.mp4

**Comment by maxceem** [comment]
Challenge https://www.topcoder.com/challenges/f9030ea9-8a63-437a-9957-8013bd8e91ea has been created for this ticket.<br/><br/>```This is an automated message for maxceem via Topcoder X```

**Comment by maxceem** [comment]
Challenge https://www.topcoder.com/challenges/f9030ea9-8a63-437a-9957-8013bd8e91ea has been assigned to xxcxy.<br/><br/>```This is an automated message for maxceem via Topcoder X```

**Comment by maxceem** [comment]
Payment task has been updated: https://www.topcoder.com/challenges/f9030ea9-8a63-437a-9957-8013bd8e91ea
*Payments Complete*
Winner: xxcxy
Copilot: maxceem
Challenge `f9030ea9-8a63-437a-9957-8013bd8e91ea` has been paid and closed.<br/><br/>```This is an automated message for maxceem via Topcoder X```

**Comment by maxceem** [comment]
Thank you @xxcxy it works now, and payment has been processed.
